rpc/rpc-core/src/main/java/com/strategyobject/substrateclient/rpc/core/decoders/GenericArrayDecoder.java

```diff
@@ -0,0 +1,24 @@
+package com.strategyobject.substrateclient.rpc.core.decoders;
+
+import com.google.common.base.Preconditions;
+import com.strategyobject.substrateclient.rpc.core.DecoderPair;
+import com.strategyobject.substrateclient.transport.RpcObject;
+import lombok.val;
+
+public class GenericArrayDecoder extends AbstractDecoder<Object[]> {
```

Comment:
The prefix `Generic` sounds confusing to me - it looks like the decoder dedicated to arrays of generics.

Reply:
agree. I was also doubtful on this